### PR TITLE
[seal]Updated to version 3.6.6. Replaced CONTROL with manifest file.

### PR DESCRIPTION
**Pull request by WeiDaiWD**
**Describe the pull request**

- #### What does your PR fix?  
  Updated `ports/seal` to version 3.6.6. Added a new feature "hexl" that depends on the `ports/hexl`. Replaced `CONTROL` file with `vcpkg.json` manifest file. Removed deprecated vcpkg commands.
  Fixed `ports/hexl` by adding another patch. Previously "hexl" does not look for its dependency in config file, causing downstream linking failures.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  x64-windows/x86-windows are not supported as the library only supports static linkage in Windows.
  No change to the CI baseline.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  I've tried my best.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: versions/s-/seal.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "b15a8a466694df183d4d36ce369f59d93d3cd326",
+      "version-string": "3.6.6",
```
Comment (JonLiu1993) [inline versions/s-/seal.json:5]: we should use "version-semver" instead of "version-string"

File: ports/seal/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "seal",
+  "version-string": "3.6.6",
```
Comment (JonLiu1993) [inline ports/seal/vcpkg.json:3]: @WeiDaiWD ,Thanks for your pr,we should use "version-semver" instead of "version-string"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@WeiDaiWD,Have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？

File: ports/seal/vcpkg.json (hunk 2 above)
Comment (WeiDaiWD) [reply ports/seal/vcpkg.json:3]: What's the difference between `"version-semver"` and `"version"`? I was told to use `"version"` when adding/updating other ports, and wonder how to decide in future.

File: versions/s-/seal.json (hunk 1 above)
Comment (WeiDaiWD) [reply versions/s-/seal.json:5]: Should I change all appearances of `"version-string"` in this file? I mean for all versions? And is it supposed to be the same with whatever used in `/ports/seal/vcpkg.json`?

**Comment by WeiDaiWD** [comment]
> @WeiDaiWD,Have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？

x64-linux and x64-windows-static work fine locally. x64-windows throws errors. `seal` does not support dynamic linking on windows, neither does `hexl`.

File: versions/s-/seal.json (hunk 1 above)
Comment (JonLiu1993) [reply versions/s-/seal.json:5]: In fact, I just pointed out the modification. You only need to modify the "version-string" to "version-semver" in vcpkg.json, and the versions/s-/seal.json file will be automatically modified

File: ports/seal/vcpkg.json (hunk 2 above)
Comment (JonLiu1993) [reply ports/seal/vcpkg.json:3]: Please refer to the official [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/specifications/versioning.md#22-package-versions)

**Comment by JonLiu1993** [comment]
If you have modified everything, please run the "./vcpkg x-add-version seal --overwrite-version " command and resubmit the changes

**Review by JonLiu1993 — APPROVED** [review]

File: ports/seal/portfile.cmake [hunk 3]
```diff
@@ -3,49 +3,39 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
-    ms-gsl SEAL_USE_MSGSL
-    zlib SEAL_USE_ZLIB
-    zstd SEAL_USE_ZSTD
-
+        ms-gsl SEAL_USE_MSGSL
+        zlib SEAL_USE_ZLIB
+        zstd SEAL_USE_ZSTD
```
Comment (ras0219-msft) [inline ports/seal/portfile.cmake:15]: I see no feature macro added to respect the `hexl` feature; perhaps you meant to add something like
```suggestion
        zstd SEAL_USE_ZSTD
    INVERTED_FEATURES
        hexl CMAKE_DISABLE_FIND_PACKAGE_HEXL
```

**Comment by ras0219-msft** [comment]
Thanks for the PR!

File: ports/seal/portfile.cmake (hunk 3 above)
Comment (WeiDaiWD) [reply ports/seal/portfile.cmake:15]: Thanks for catching this. I've added `SEAL_USE_INTEL_HEXL` to the latest commits.

**Comment by WeiDaiWD** [comment]
Do not merge. Debugging a linking error.

**Comment by WeiDaiWD** [comment]
> Do not merge. Debugging a linking error.

This is resolved and tested locally.

**Comment by WeiDaiWD** [comment]
I've also edited the PR description to cover changes to `ports/hexl`.

File: ports/seal/portfile.cmake (hunk 3 above)
Comment (JonLiu1993) [reply ports/seal/portfile.cmake:15]: @WeiDaiWD,INVERTED_FEATURES No need?

Comment (WeiDaiWD) [reply ports/seal/portfile.cmake:15]: Nope, it is controlled by a single SEAL_USE_XXX argument.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks!

**Pull request merged**